packages/synapse-react-client/src/components/dataaccess/UseAccessRequirementTable.tsx

```diff
@@ -0,0 +1,236 @@
+    header: props => <ColumnHeader {...props} title={'Created On'} />,
+    cell: ({ getValue }) => formatDate(dayjs(getValue())),
+    enableSorting: true,
+  }),
+]
+const DEFAULT_SORTING_STATE: SortingState = [
+  {
+    desc: true,
+    id: 'createdOn',
+  },
+]
+type UseAccessRequirementTableOptions = {
+  nameOrID?: string
+  relatedProjectId?: string
+  reviewerId?: string
+  accessType?: ACCESS_TYPE
+}
+
+export function useAccessRequirementTable(
+  opts: UseAccessRequirementTableOptions,
+) {
+  const { nameOrID, relatedProjectId, reviewerId, accessType } = opts
+  const [sortingState, setSortingState] = useState<SortingState>(
+    DEFAULT_SORTING_STATE,
+  )
+
+  // Map the @tanstack/react-table state object to the Synapse API object
+  const apiRequestSortState: AccessRequirementSearchRequest['sort'] =
+    useMemo(() => {
+      return sortingState.map((s: ColumnSort): AccessRequirementSearchSort => {
+        const direction: SortDirection = s.desc ? 'DESC' : 'ASC'
+
+        if (s.id == 'createdOn') {
+          return {
+            field: 'CREATED_ON',
+            direction,
+          }
+        } else if (s.id == 'name') {
+          return {
+            field: 'NAME',
+            direction,
+          }
+        } else {
+          throw new Error(
+            `Could not map sort field ${s.id} to AccessRequirementSortField`,
+          )
+        }
+      })
+    }, [sortingState])
+
+  const searchRequest: Omit<AccessRequirementSearchRequest, 'nextPageToken'> =
+    useMemo(() => {
+      // SWC-6615: If the input string is a single integer, assume it's the AR ID.  Otherwise, use as the nameContains field.
+      let nameContains: string | undefined = undefined
+      let ids: number[] | undefined = undefined
+      if (nameOrID !== undefined) {
+        const nameOrIDTrimmed = nameOrID.trim()
+        if (isIntegerInput(nameOrIDTrimmed)) {
+          ids = [Number.parseInt(nameOrIDTrimmed)]
+        } else {
+          nameContains = nameOrIDTrimmed
+        }
+      }
+      return {
+        ids,
+        nameContains,
+        relatedProjectId,
+        reviewerId,
+        accessType,
+        sort: apiRequestSortState,
+      }
+    }, [
+      nameOrID,
+      relatedProjectId,
+      reviewerId,
+      accessType,
+      apiRequestSortState,
+    ])
+
+  const { data, hasNextPage, fetchNextPage, isLoading } =
+    useSearchAccessRequirementsInfinite(searchRequest)
+
+  const accessRequirements = useMemo(
+    () => data?.pages.flatMap(page => page.results) ?? [],
+    [data],
+  )
+
+  const table: Table<AccessRequirementSearchResult> =
+    useReactTable<AccessRequirementSearchResult>({
+      data: accessRequirements,
+      columns: columns,
+      state: {
+        sorting: sortingState,
+      },
+      onSortingChange: setSortingState,
+      getRowId: row => row.id,
+      getCoreRowModel: getCoreRowModel(),
+      columnResizeMode: 'onChange',
+      manualSorting: true,
+      enableMultiSort: false,
```

Comment:
We could easily toggle multi-sort on, but for now it's not very useful (the two existing sort fields, name and createdOn, never have duplicates) and it just makes it harder to reset your sort state.